abci/types/application.go

```diff
@@ -98,11 +101,18 @@ func (BaseApplication) ApplySnapshotChunk(req RequestApplySnapshotChunk) Respons
 }
 
 func (BaseApplication) PrepareProposal(req RequestPrepareProposal) ResponsePrepareProposal {
-	return ResponsePrepareProposal{BlockData: req.BlockData}
+	squareSize := len(req.Txs)
```

Comment:
[question] why is the squareSize set to the number of transactions? I think of these as two distinct values. For example, a block with 100 send transactions could have a square size of 4.

Reply:
[optional] Thanks for clarifying that they are placeholders. If they are expected to be overwritten by celestia-app, then proposal to do something like:

```go
func (BaseApplication) PrepareProposal(req RequestPrepareProposal) ResponsePrepareProposal {
	placeholderDataHash := tmhash.Sum(nil)
	placeholderSquareSizeBytes := make([]byte, 8)
	binary.BigEndian.PutUint64(placeholderSquareSizeBytes, uint64(1))

	// placeholderDataHash and placeholderSquareSizeBytes are expected to be
	// overwritten by celestia-app.
	req.Txs = append(req.Txs, placeholderDataHash, placeholderSquareSizeBytes)
	return ResponsePrepareProposal{Txs: req.Txs}
}
```